dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/AdvancedSearchPage.java

```diff
@@ -100,6 +101,7 @@ public AdvancedSearchPage(DataverseDao dataverseDao, DatasetFieldServiceBean dat
 
     // -------------------- LOGIC --------------------
 
+    @PostConstruct
     public void init() {
```

Comment:
When we are entering the page in a classic way (using link from homepage). Will this code execute once or twice? There is also code that executes this method in `advanced.xhtml` file: `<f:viewAction action="#{AdvancedSearchPage.init}"/>`

Reply:
Good catch. It was indeed called twice. I removed the `viewAction` as now in all the cases `@PostConstruct` is called reliably for the `@ViewScoped` bean.